napari/utils/_tests/test_theme.py

```diff
@@ -30,6 +31,13 @@ def test_get_theme():
     assert isinstance(theme, Theme)
 
 
+def test_get_system_theme(monkeypatch):
```

Comment:
Added this test due to code cov.
Not sure what would be better. Basically `system` is a dummy theme, not a real theme: it's either `light` or `dark`.

Reply:
Ok, that makes a lot of sense. The system to real theme conversion is kinda checked here as well:
https://github.com/napari/napari/blob/0478143d6d90e831d1727515dc1ece6fe96333e8/napari/components/_tests/test_viewer_keybindings.py#L32-L37
I could make this test be:
```python
def test_theme_toggle_from_system_theme(): 
     get_settings().appearance.theme = 'system' 
     viewer = ViewerModel() 
     actual_theme = get_system_theme()
     assert viewer.theme = actual_theme
     toggle_theme(viewer)
     # ensure that theme has changed
     assert not viewer.theme == actual_theme 
     toggle_theme(viewer)
     # ensure we have looped back to whatever system was
     assert viewer.theme == actual_theme 
```
Then delete the test in `test_theme.py`?

Reply:
I like it! Since that test is concerned about the toggle behavior, I think it makes sense to toggle twice and assert what's expected.

Though the first `assert` in the new code needs to use `==` instead of `=`.

Reply:
Grrr. Pooped on, the test fails, because:
```
In [1]: import napari
In [2]: from napari.components.viewer_model import ViewerModel
In [3]: viewer1 = ViewerModel()
In [4]: viewer1.theme
Out[4]: 'system'

In [5]: viewer2 = napari.Viewer()
In [6]: viewer2.theme
Out[6]: 'dark'

In [7]: from napari.settings import get_settings
In [8]: get_settings().appearance.theme
Out[8]: 'system'
```
with `ViewerModel` the theme is returned as `system`, which isn't an actual theme, meaning `get_system_theme` wasn't run. If you use a real napari viewer, it works. 
So I think this test needs qt? Or it's something in ViewerModel. 
I'm definitely out of my depth now...

Reply:
Like I said originally, the change is optional. Feel free to keep the original test or remove it. Either way, this PR shouldn't be blocked by it.

Reply:
So the root cause is that `ViewerModel` uses this function:
https://github.com/napari/napari/blob/e4591ca99157f03af2f34ad8b154e9358e74e31b/napari/components/viewer_model.py#L87-L88
Which just checks settings and doesn't actually ensure an actual theme.
So I think this is an oversight in this PR and should be addressed?
I think that there should be a check there, along the lines of:
```
if _current_theme == "system":
                 _current_theme  = get_system_theme()
```
Although to be honest I can't say i'm totally clear on ViewerModel and viewer.

Reply:
I guess this comment thread snowballed... let me know if I can help in any way!